pennylane/compiler/__init__.py

```diff
@@ -146,23 +153,67 @@ def circuit(x: int):
 >>> circuit(5)
 array(1.)
 
+Note that AutoGraph results in additional
+restrictions, in particular whenever global state is involved. Please refer to the
+:doc:`AutoGraph guide <catalyst:dev/autograph>`
+for a complete discussion of the supported and unsupported use-cases.
+
 For more details on using the :func:`~.qjit` decorator and Catalyst
 with PennyLane, please refer to the Catalyst
-`quickstart guide <https://docs.pennylane.ai/projects/catalyst/en/latest/dev/quick_start.html>`__,
-as well as the `sharp bits and debugging tips <https://docs.pennylane.ai/projects/catalyst/en/latest/dev/sharp_bits.html>`__
+:doc:`quickstart guide <catalyst:dev/quick_start>`,
+as well as the :doc:`sharp bits and debugging tips <catalyst:dev/sharp_bits>`
 page for an overview of the differences between Catalyst and PennyLane, and
 how to best structure your workflows to improve performance when
 using Catalyst.
 
 Adding a compiler
 -----------------
 
-For any compiler packages seeking to be registered, it is imperative that they expose the 'entry_points' metadata
-under the designated group name: ``pennylane.compilers`` with the following entry points:
+For any compiler packages seeking to be registered, it is imperative that they
+expose the ``entry_points`` metadata under the designated group name
+``pennylane.compilers``, with the following entry points:
+
+- ``context``: Path to the compilation evaluation context manager.
+  This context manager should have the method ``context.is_tracing()``,
+  which returns ``True`` if called within a program that is being traced
+  or captured.
+
+- ``ops``: Path to the compiler operations module. This operations module
+  may contain compiler specific versions of PennyLane operations,
+  for example such as :func:`~.cond`, :func:`~.measure`, and :func:`~.adjoint`.
+  Within a JIT context, PennyLane operations may dispatch to these functions.
+
+- ``qjit``: Path to the JIT compiler decorator provided by the compiler.
+  This decorator should have the signature ``qjit(fn, *args, **kwargs)``,
+  where ``fn`` is the function to be compiled.
+
+In order to support the following two syntaxes,
+
+.. code-block:: python
+
+    @qml.qjit
+    def function(x, y):
+        ...
+
+    @qml.qjit(verbose=True, additional_args, ...)
+    def function(x, y):
+        ...
+
+where the ``qjit`` decorator can be applied with and without arguments,
+you should ensure that your ``qjit`` decorator itself returns a decorator
+if no function is provided:
+
+.. code-block:: python
+
+    def qjit(fn=None, **kwargs):
+        if fn is not None:
+            return compile_fn(fn, **kwargs)
+
+        def wrapper_fn(fn):
+            return compile_fn(fn, **kwargs)
+
+        return wrapper_fn
```

Comment:
@maliasadi I fleshed out this section a bit more -- was thinking in terms of 'if I have no idea how to make my 3rd party compiler compatible, what additional information would I need?'

Reply:
Looks great! Thanks @josh146 :100: